go/kbfs/libkbfs/folder_block_ops.go

```diff
@@ -1485,6 +1487,25 @@ func (fbo *folderBlockOps) updateEntryLocked(ctx context.Context,
 	} else {
 		_ = fbo.makeDirDirtyLocked(lState, parentPath.TailPointer(), unrefs)
 	}
+
+	// If we're in the middle of syncing the directories, but the
+	// current file is not yet being synced, we need to re-apply this
+	// update after the sync is done, so it doesn't get lost after the
+	// syncing directory block is readied.  This only applies to dir
+	// updates being caused by file changes; other types of dir writes
+	// are protected by `folderBranchOps.syncLock`, which is held
+	// during `SyncAll`.
+	if fbo.dirtyDirsSyncing && !fbo.doDeferWrite {
```

Comment:
Is there a race whereby `fbo.dirtyDirsSyncing` is `false` here, but `true` when the update would normally happen above, and so it gets wiped out but we don't notice?

Reply:
No, `fbo.dirtyDirsSyncing` is protected by the same lock that is required for calling this function, so it can't change between the above code and here.  (If you're asking something else, let me know -- not sure if I understood the question completely.)

Reply:
I think I slightly misunderstood the purpose of `dirtyDirsSyncing` when I was writing this, and am now convinced the race that I was thinking of does not exist.